pkg/apiserver/apiserver.go

```diff
@@ -99,13 +98,6 @@ func (cfg *Config) Complete() CompletedConfig {
 }
 
 func (config completedConfig) New() (*ClusterPediaServer, error) {
-	if config.ClientConfig == nil {
-		return nil, fmt.Errorf("CompletedConfig.New() called with config.ClientConfig == nil")
-	}
-	if config.StorageFactory == nil {
-		return nil, fmt.Errorf("CompletedConfig.New() called with config.StorageFactory == nil")
-	}
```

Comment:
This is necessary because if these two fields are not set in config, the caller will panic.
Also, the fields in Config are public fields and it cannot be assumed that the caller has set them